File: support-backend/app/api/v1/endpoints/audit.py
"""Admin audit-log viewer endpoints."""

from typing import Optional
from uuid import UUID

from fastapi import APIRouter, Depends, Query

from app.db.repositories.audit_repo import AuditRepo
from app.dependencies import get_admin_user
from app.schemas.common import build_pagination

router = APIRouter()


@router.get("")
async def list_audit_logs(
    page: int = 1,
    page_size: int = Query(50, ge=1, le=200),
    actor_id: Optional[UUID] = None,
    action: Optional[str] = None,
    resource_type: Optional[str] = None,
    project_id: Optional[UUID] = None,
    admin: dict = Depends(get_admin_user),
) -> dict:
    """List audit-log entries with optional filters (admin only).

    Supports filtering by actor, action, resource type, and project, returned
    newest-first with pagination. Each entry embeds the acting user's basic
    profile for display.
    """
    page = max(page, 1)
    repo = AuditRepo()
    logs, total = await repo.list_with_filters(
        page=page,
        page_size=page_size,
        actor_id=actor_id,
        action=action,
        resource_type=resource_type,
        project_id=project_id,
    )
    return {
        "data": logs,
        "pagination": build_pagination(total, page, page_size).model_dump(),
    }

File: support-frontend/src/types/audit.ts
/**
 * Audit-log TypeScript types.
 */

import type { PaginationMeta } from "@/types/user";

/** Basic profile of the user who performed an audited action. */
export interface AuditActor {
  id: string;
  name: string;
  email: string;
}

/** A single audit-log entry as returned by the backend. */
export interface AuditLog {
  id: string;
  actor_id: string;
  actor: AuditActor | null;
  action: string;
  resource_type: string;
  resource_id: string;
  project_id: string | null;
  metadata: Record<string, unknown>;
  ip_address: string | null;
  created_at: string;
}

/** Optional filters for the audit-log list request. */
export interface AuditLogFilters {
  action?: string;
  resource_type?: string;
  actor_id?: string;
  project_id?: string;
}

export interface AuditLogListResponse {
  data: AuditLog[];
  pagination: PaginationMeta;
}
